pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -562,32 +559,39 @@ func aceToRules(bridgeName string, vifName string, ace types.ACE, ipVer int, bri
 				log.Errorln(errStr)
 				return nil, errors.New(errStr)
 			}
-			if appIP == "" {
+			if appNetInfo.AppIP == "" {
 				errStr := fmt.Sprintf("PortMap without appIP for lport %s/protocol %s: %+v",
 					lport, protocol, ace)
 				log.Errorln(errStr)
 				return nil, errors.New(errStr)
 			}
 			targetPort := fmt.Sprintf("%d", action.TargetPort)
-			target := fmt.Sprintf("%s:%d", appIP, action.TargetPort)
-			// These rules should only apply on the uplink
-			// interfaces but for now we just compare the protocol
-			// and port number.
-			// The DNAT/SNAT rules do not compare fport and ipset
-			rule1 := []string{"PREROUTING",
-				"-p", protocol, "--dport", lport,
-				"-j", "DNAT", "--to-destination", target}
-			// Make sure packets are returned to zedrouter and not
-			// e.g., out a directly attached interface in the domU
-			rule2 := []string{"POSTROUTING",
-				"-p", protocol, "-o", bridgeName,
-				"--dport", targetPort, "-j", "SNAT",
-				"--to-source", bridgeIP}
+			target := fmt.Sprintf("%s:%d", appNetInfo.AppIP, action.TargetPort)
+			// These rules are applied on the upLink
+			// interfaces and port number.
+			// loop through the uplink interfaces
+			for _, upLink := range appNetInfo.UpLinks {
+				log.Infof("upLink %s\n", upLink)
+				// The DNAT/SNAT rules do not compare fport and ipset
+				// Make sure packets are returned to zedrouter and not
+				// e.g., out a directly attached interface in the domU
+				rule1 := []string{"PREROUTING",
+					"-i", upLink, "-p", protocol,
+					"--dport", lport, "-j", "DNAT",
+					"--to-destination", target}
+				rule2 := []string{"POSTROUTING",
+					"-o", upLink, "-p", protocol,
```

Comment:
Isn't the output link network instance bridge interface? Shouldn't the output interface match condition be "-o bridgeName"?

Reply:
yes. bridge interface is correct. will change.